### PR TITLE
libpng: add with_zlibng option

### PR DESCRIPTION
**Pull request by Nekto89**
Specify library name and version:  **libpng/all**

Allow using zlib-ng instead of zlib.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by RubenRBS** [comment]
Hi @Nekto89 thanks a lot for your contribution. We see the value on having this option available for recipes that depend on Zlib. The only problem is to ensure a homogenous API for every recipe for ease of use.
I've added this point to our next CCI team meeting in a couple of days, will let you know once we determine the best way forward :)

**Comment by Nekto89** [comment]
> Hi @Nekto89 thanks a lot for your contribution. We see the value on having this option available for recipes that depend on Zlib. The only problem is to ensure a homogenous API for every recipe for ease of use. I've added this point to our next CCI team meeting in a couple of days, will let you know once we determine the best way forward :)

I use a bunch of libraries that use zlib and recently discovered that zlib-ng improves performance a lot. So replacing zlib with zlib-ng in compatibility mode looks like a free win. I'm looking forward to your decision.

**Comment by Nekto89** [comment]
> Hi @Nekto89 thanks a lot for your contribution. We see the value on having this option available for recipes that depend on Zlib. The only problem is to ensure a homogenous API for every recipe for ease of use. I've added this point to our next CCI team meeting in a couple of days, will let you know once we determine the best way forward :)

Hi @RubenRBS . Were you able to discuss the issue with the team?

File: recipes/libpng/all/conanfile.py [hunk 1]
```diff
@@ -26,6 +27,7 @@ class LibpngConan(ConanFile):
         "sse": [True, False],
         "vsx": [True, False],
         "api_prefix": ["ANY"],
+        "with_zlibng": [True, False],
```
Comment (valgur) [inline recipes/libpng/all/conanfile.py:30]:
```suggestion
        "with_zlib": ["zlib", "zlibng"],
```
Maybe future-proof it a bit.

**Pull request closed without merging**